From 0e9c555347e4790863b072d114c5226851cad4e0 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Tue, 28 Feb 2023 11:22:22 +0100
Subject: [PATCH 0264/1141] arm64: dts: st: enable management of the Cortex-M33
 on stm32mp257f-ev

Enable the IPCC and the remoteproc for the management of the cCortex-M33.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I87c3e1c17f07e69631f92956f6a477b83ebeb825
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/310433
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev.dts
index 010c8b4a13a6..0eef86003ff8 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev.dts
@@ -153,6 +153,18 @@ &i2c2 {
 	/delete-property/dma-names;
 };
 
+&ipcc1 {
+	status = "okay";
+};
+
+&m33_rproc {
+	mboxes = <&ipcc1 0x100>, <&ipcc1 0x101>, <&ipcc1 2>;
+	mbox-names = "vq0", "vq1", "shutdown";
+	memory-region = <&ipc_shmem_1>, <&vdev0vring0>,
+			<&vdev0vring1>, <&vdev0buffer>;
+	status = "okay";
+};
+
 &pcie_rc {
 	pinctrl-names = "default", "init";
 	pinctrl-0 = <&pcie_pins_a>;
-- 
2.39.2

